src/core/ext/filters/client_channel/lb_policy/ring_hash/ring_hash.cc

```diff
@@ -314,16 +314,17 @@ RingHash::Ring::Ring(RingHash* parent,
     const ServerAddressWeightAttribute* weight_attribute = static_cast<
         const ServerAddressWeightAttribute*>(sd->address().GetAttribute(
         ServerAddressWeightAttribute::kServerAddressWeightAttributeKey));
+    auto addr_str = grpc_sockaddr_to_string(&sd->address().address(), false);
     AddressWeight address_weight;
-    address_weight.address =
-        grpc_sockaddr_to_string(&sd->address().address(), false);
+    address_weight.address = std::string(addr_str.value());
     if (weight_attribute != nullptr) {
       GPR_ASSERT(weight_attribute->weight() != 0);
       address_weight.weight = weight_attribute->weight();
     }
     sum += address_weight.weight;
     address_weights.push_back(std::move(address_weight));
   }
+  GPR_ASSERT(sum > 0);
```

Comment:
I don't think we need to add this assertion anymore.

Reply:
Done.